src/Tasks/Microsoft.NET.Build.Tasks/ResourceUpdater.cs

```diff
@@ -169,6 +172,45 @@ protected override bool ReleaseHandle()
         /// </summary>
         private readonly SafeUpdateHandle hUpdate;
 
+        ///<summary>
+        /// Determines if the ResourceUpdater is supported by the current operating system.
+        /// Some versions of Windows, such as Nano Server, do not support the needed APIs.
+        /// </summary>
+        public static bool IsSupportedOS()
+        {
+            if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                return false;
+            }
+
+            var module = Kernel32.LoadLibraryEx("kernel32.dll", IntPtr.Zero, 0);
+            if (module == IntPtr.Zero)
+            {
+                return false;
+            }
+
+            // Ensure the export exists before P/Invoking
+            var proc = Kernel32.GetProcAddress(module, "BeginUpdateResourceW");
+            Kernel32.FreeLibrary(module);
+            if (proc == UIntPtr.Zero)
+            {
+                return false;
+            }
+
+            // On Nano Server 1709+, the export resolves but returns a null handle with a zero error
+            // Try to call `BeginUpdateResource` with an invalid parameter; the error should be non-zero if supported
+            SafeUpdateHandle handle = Kernel32.BeginUpdateResource("", false);
```

Comment:
Is this the right way to probe this?  Have we asked the right folks in Windows?  Is it OK to call BeginUpdateResource without calling EndUpdateResource?

Reply:
Do you have a contact in Windows to query that might know?  This is relatively hacky, so I'd love a proper way of doing this.  Ideally (and what I assumed would be the case) Nano Server wouldn't export functions that aren't implemented as that's the simplest way to check if we can use an API.

Re: calling `EndUpdateResource`: it will from the `SafeUpdateHandle`, but this needs a `using` statement, which I'll fix.